### PR TITLE
[13.0] [IMP] purchase_stock_picking_invoice_link: Use the same technique as in stock_picking_invoice_link

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT38550

please @pedrobaeza @victoralmau review this

### CODE REVIEW

**Review by victoralmau — APPROVED** [review]
According to the logic of `stock_picking_invoice_link` it seems to make sense and everything is correct.

Remember to also apply these changes to v14 as well.

File: purchase_stock_picking_invoice_link/models/stock_move.py [hunk 1]
```diff
@@ -8,6 +8,8 @@ class StockMove(models.Model):
     _inherit = "stock.move"
 
     def write(self, vals):
+        """Method used to associate the stock.move with the created account.move.line
```
Comment (pedrobaeza) [inline purchase_stock_picking_invoice_link/models/stock_move.py:11]: For what is this needed?

**Review by pedrobaeza — COMMENTED** [review]
Is the case of the credit note with "cancel + new invoice" option covered in this module?

File: purchase_stock_picking_invoice_link/models/stock_move.py (hunk 1 above)
Comment (CarlosRoca13) [reply purchase_stock_picking_invoice_link/models/stock_move.py:11]: Seeing this test, It's necesary to cover when we invoice the qty demanded before receive products from supplier. But the stock moves have to be linked with the corresponding invoice line.

https://github.com/OCA/stock-logistics-workflow/blob/92d19c57be418f1e53b773c4de232679808fc45a/purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py#L44-L68

**Comment by CarlosRoca13** [comment]
> Is the case of the credit note with "cancel + new invoice" option covered in this module?

Yes, it is. I added a test to check it too

File: purchase_stock_picking_invoice_link/models/stock_move.py (hunk 1 above)
Comment (pedrobaeza) [reply purchase_stock_picking_invoice_link/models/stock_move.py:11]: OK, then please expand the docstring to say that the invoice was done before receiving the product. And docstrings end with the triple quote isolated in a new line if multi-line. See PEP257 for more information: https://peps.python.org/pep-0257/

**Comment by pedrobaeza** [comment]
> > Is the case of the credit note with "cancel + new invoice" option covered in this module?
> 
> Yes, it is. I added a test to check it too

I don't see such test

**Comment by CarlosRoca13** [comment]
> > > Is the case of the credit note with "cancel + new invoice" option covered in this module?
>>
> > Yes, it is. I added a test to check it too
>
> I don't see such test

https://github.com/OCA/stock-logistics-workflow/pull/1068/files#diff-a6f3c01f07d15948730d70d15d1f5f925d2677eb8fbeb0e71123c59e3e1b26aeR67-R97

File: purchase_stock_picking_invoice_link/models/stock_move.py (hunk 1 above)
Comment (CarlosRoca13) [reply purchase_stock_picking_invoice_link/models/stock_move.py:11]: Changes done

File: purchase_stock_picking_invoice_link/models/stock_move.py [hunk 2]
```diff
@@ -8,6 +8,10 @@ class StockMove(models.Model):
...
+        when the invoicing method of the product is 'purchase' and the invoice is done
+        before receive the products.
```
Comment (pedrobaeza) [inline purchase_stock_picking_invoice_link/models/stock_move.py]:
```suggestion
        before receiving the products.
```

**Comment by pedrobaeza** [comment]
> > > > Is the case of the credit note with "cancel + new invoice" option covered in this module?
> > > 
> > > 
> > > Yes, it is. I added a test to check it too
> > 
> > 
> > I don't see such test
> 
> https://github.com/OCA/stock-logistics-workflow/pull/1068/files#diff-a6f3c01f07d15948730d70d15d1f5f925d2677eb8fbeb0e71123c59e3e1b26aeR67-R97

That's not the case I'm talking about. I'm talking about selecting this option in the "Add Credit Note" wizard:

Selección_007

**Comment by CarlosRoca13** [comment]
@pedrobaeza test added and change done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please fw-port the changes.

/ocabot merge major

**Pull request merged**